src/cpp/dpm_sampler.cpp

```diff
@@ -44,15 +45,59 @@
 #include <nmmintrin.h>
 #endif
 
-#include <gsl/gsl_randist.h>
-#include <gsl/gsl_rng.h>
-#include <gsl/gsl_sf_exp.h>
-#include <gsl/gsl_sf_log.h>
-#include <gsl/gsl_sf_pow_int.h>
-
 #include "data_structures.hpp"
 #include "dpm_sampler.hpp"
 
+#include <random>
+#include <boost/random/beta_distribution.hpp>
+std::mt19937 rg;
```

Comment:
does `rg` have to have external linkage? Why not `static` (or better yet, an unnamed namespace, let's use C++ features)

Reply:
Also, is this `rg` being shared between thread? If so, this can't work, as rngs arent thread-safe in C++

Reply:
`#include <nmmintrin.h>` since when do you fancy sse instructions, David?

Reply:
That was from Susi's `POPCNT` optimization, which enables O(1) calculation of hamming distance ;)

Reply:
Lol, including one header for one popcnt :D

Reply:
Regarding threading: it's not done within diri_sampler it'self.
It's the overall shorah python codes which launches multiple process in parallel instead. Several windows get diri_sampler-ed concurrently.

`POPCNT` fun thing: LLVM's clang's optimizer is able to recognize the long form Donald Knuth's algorithm, and correctly emit a POPCNT instruction in machine code. 
Try it in [godblot.org](https://godbolt.org/), it0s fun to watch.

Reply:
Still, please stuff as many things into unnamed namespaces as possible, preventing symbol pollution